keps/sig-network/2079-network-policy-port-range/README.md

```diff
@@ -232,20 +232,40 @@ _This section must be completed when targeting alpha to a release._
 
 * **Can the feature be disabled once it has been enabled (i.e. can we roll back
   the enablement)?**
-  Yes, but CNIs relying on the new field wont recognize it anymore
+  Yes, fields created before feature gate being disabled will not be dropped unless the user removes it from the object.
```

Comment:
This isn't exactly true - if the feature is disabled there shouldn't be a way to drop the field even...

Reply:
Hum, okay, thinking here loudly (sorry for being too verbose!) 

By the impl, when we edit an object that have the NetPol EndPort, but with the FG disabled, the EndPort field appears there (like in kubectl edit)

When I remove the field, PrepareForUpdate will run, and skip the dropEndPort because the field is in use, so it will follow its normal flow. (https://github.com/kubernetes/kubernetes/blob/f0601ac64d39c601a66f8f6e6f69d55995c2b32a/pkg/registry/networking/networkpolicy/strategy.go#L62-L64)

The normal flow is to take the old object, and if it's changed add +1 into Generation and then apply the desired changes.

So probably I'll need to also add into something after the PrepareForUpdate to not drop the EndPort field, even if the user has removed it?

Reply:
This is a tricky topic - I think I didn't realize one thing when writing this comment.

Looking into other examples across the codebase, the only thing we're doing is we're preventing from setting the field when the FG is switched on and silently dropping it if it was set. We allow to explicitly dropping it at any time.


So, I just actually suggest a simple nit:
```
Yes. One caveat here is that NetworkPolicies created with EndPort fiel set when the feature was enabled will continue to have that field set when the feature is disabled unless user removes it from the object.
```

[The sentence below kind-of duplicates the message]

Reply:
Great :)

I've improved the sentence here consolidating your suggestion and improving mine (I don't know where was my head when  I wrote the next sentence, but it took me 3 reads to understand what I wanted to say).

Made another commit here